### ISSUE TITLE
Budgets as negative numbers

### ISSUE THREAD

**Issue by giuseppebaldi** [soloam/ha-fireflyiii-integration#40]
Question: I noticed that the budget entity states are brought in as negative numbers but the budget limit attribute is a positive number, is there a reason for this? I am trying to use the [Bar Card](https://github.com/custom-cards/bar-card) to set up some budget visuals in HA but the negative/positive numbers are making it difficult (I wanted to use the budget limit as the max limit so I wouldn't have to manually enter these). Do you have any suggestions a workaround? Or can the budget and budget limit both be positive numbers (or both negative would work also)?

Also, would it be possible to bring the total budget & total spent from Firefly?

**Comment by soloam** [comment]
Hello @giuseppebaldi!

The first idea for the budget sensor was to show the amount you would need to get to your objective... or if you are past it! So if you have somting on the budget, you will need to  "add" money to make it full. Ate the moment I'm working on the next release, I'll probably add a checkbox to the config to make the budgets positive and not negative.

Regarding you next question, I'm adding the stats form the main page to sensors, and will be out in the next release! Is that it?

Thank you

**Comment by giuseppebaldi** [comment]
I see, that makes sense. A checkbox would be a great solution. The stats from the main page would be terrific to but I was referring to the total budget and total spent from the Budgets page, see the attached screenshot.

Thank you so much!
![Screenshot 2024-11-20 180058](https://github.com/user-attachments/assets/b83eb80c-8801-40e0-8cdf-cb19e894646a)

**Comment by soloam** [comment]
That information isn't already on the attributes of the sensor? If not probably I'm looking at the in dev release! I'll try to get it out as quickly as possible!

**Comment by giuseppebaldi** [comment]
I'm sorry, what I meant is the total of all the budgets added up into one sensor, along with all the limits added up and what has been spent. I have a sensor for each budget item (groceries, utilities, etc) but would like one for the total budget. I hope that make sense. Thank you!

**Comment by giuseppebaldi** [comment]
Yes, I considered that but would have to remember to update it if I remove or add any budgets (which really wouldn't be a big deal).

**Comment by soloam** [comment]
You don't need to! You could create a sensor with a Jinja mapping. Something like this:

```
{{ states.sensor|selectattr('attributes.fireflyiii_type', 'eq', 'piggy_banks')|map(attribute='state')|map('float')|sum }} 
```

You are selecting all sensors with the "fireflyiii_type" equal to "piggy_banks" and mapping it's state to a float and in the end making a sum!

**Comment by giuseppebaldi** [comment]
Can you possibly give me some insight into why this doesn't work (based of your previous example)?

`{{ states.sensor|selectattr('attributes.fireflyiii_account_type', 'eq', 'asset_accounts')|map(attribute='state')|map('float')|sum }} `

I am trying to create a sensor that would sum all asset accounts similar to the budgets you helped me with before.

TIA
